### PR TITLE
feat: stack reconstruct

### PR DESCRIPTION
**Pull request by lightsing**
### Changes
- env `CHECK_MEM_STRICT` changes to `CHECK_MEM_STACK_LEVEL`

### CODE REVIEW

**Comment by lispc** [comment]
<del>after this finished, we can bench some ccc time. i guess deserialize time will be reduced a lot </del>

oh this can be next PR. Add a stack_head to trace struct.

File: bus-mapping/src/circuit_input_builder/access.rs [hunk 1]
```diff
@@ -119,6 +120,30 @@ impl AccessSet {
...
+        for trace in geth_data.geth_traces.iter() {
+            access_set.extend_from_traces(trace.prestate.as_ref().unwrap());
+        }
+        access_set
+    }
+
+    pub(crate) fn from_l2_geth_data(geth_data: &GethData) -> Self {
+        let mut access_set = AccessSet::default();
+        access_set.add_account(geth_data.eth_block.author.unwrap());
+        for (addr, storage) in geth_data.block_trace.storage_trace.storage_proofs.iter() {
```
Comment (lispc) [inline bus-mapping/src/circuit_input_builder/access.rs]: you need also add geth_data.block_trace.storage_trace.proofs

**Comment by lightsing** [comment]
ping @lispc

File: zkevm-circuits/src/super_circuit/precompile_block_trace.rs [hunk 2]
```diff
@@ -831,14 +831,14 @@ pub(crate) fn block_invalid_precompile() -> BlockTrace {
...
-                .to(accs[3].address)
-                .gas(Word::from(2_000_000u64));
+            // txs[1]
+            //     .from(wallet_a.clone())
+            //     .to(accs[2].address)
+            //     .gas(Word::from(2_000_000u64));
+            // txs[2]
+            //     .from(wallet_a.clone())
+            //     .to(accs[3].address)
+            //     .gas(Word::from(2_000_000u64));
```
Comment (lispc) [inline zkevm-circuits/src/super_circuit/precompile_block_trace.rs]: what is wrong here?

**Comment by lispc** [comment]
i noticed the "tests" sub module updated in this PR. It that ok?

File: zkevm-circuits/src/super_circuit/precompile_block_trace.rs (hunk 2 above)
Comment (lightsing) [reply zkevm-circuits/src/super_circuit/precompile_block_trace.rs]: forget to remove comments

**Comment by lightsing** [comment]
> i noticed the "tests" sub module updated in this PR. It that ok?

I remember it was upgraded due to some issues of upgrading l1geth version.

**Comment by lispc** [comment]
looks good to me.   I will run testool later today then merge it.

**Comment by lispc** [comment]
@lightsing a lot failures.  I would suggest check these failed precompiles like `ecmul_1-2_616_28000_96_d0_g0_v0`.

https://circuit-release.s3.us-west-2.amazonaws.com/testool/nightly.1706707610.0cecef0.html

**Comment by lightsing** [comment]
default suit: https://circuit-release.s3.us-west-2.amazonaws.com/testool/default.1707280178.9aef6df.html
checked: all failed case are not present in test_ids.txt

**Review by lispc — APPROVED** [review]

**Pull request merged**